### PR TITLE
Watool trusted advisor integration lab

### PR DESCRIPTION
**Pull request by awsdunc**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sssalim-aws — COMMENTED** [review]
Will it make more sense to prescribe the Tag  Key and Value ?

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/1_prerequisites.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ 
+![Section1 AWS Compute Optimizer](/watool/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/Images/section1/ComputeOptimizer.png)
+ 
+## Cost Optimization checks in Amazon Trusted Advisor 
+[Amazon Trusted Advisor](https://aws.amazon.com/premiumsupport/knowledge-center/trusted-advisor-intro/) provides best practices (or checks) in four categories: cost optimization, security, fault tolerance, and performance improvement. This demo will use "Low Utilization Amazon EC2 Instances" check in Cost Optimization.
+ 
+![Section1 AWS Trusted Advisor](/watool/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/Images/section1/TA.png)
+ 
+## Attach AWS Tags to Amazon EC2 Instances 
+Amazon EC2 Instances must have AWS Tags. If you have not associated AWS Tags with Amazon EC2 Instances with any AWS Tags, [tag your Amazon EC2 resources](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html). You can use your own key and value.
```
Comment (sssalim-aws) [inline content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/1_prerequisites.md:23]: Will it make more sense to prescribe the tag ?

 ```
    Key = workload
    Value = wademo
    ```

Comment (sssalim-aws) [inline content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/1_prerequisites.md:23]: Will it make more sense to prescribe the tag ?

 ```
    Key = workload
    Value = wademo
    ```

**Review by sssalim-aws — CHANGES_REQUESTED** [review]
Please see comment on Pre-req. and fix image /Images/section5/workloadID.png to use image block rather than pen scratch.

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/1_prerequisites.md (hunk 1 above)
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/1_prerequisites.md:23]: updated as follows:
 Key = Environment
 Value = watooldemo

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+In this section, we will deploy our base lab infrastructure using [AWS Serverless Application Model (AWS SAM)](https://aws.amazon.com/serverless/sam/) in [AWS Cloud9](https://aws.amazon.com/cloud9/) environment. This will consist of a public [Amazon API Gateway](https://aws.amazon.com/api-gateway/) which connects to [AWS Lambda](https://aws.amazon.com/lambda/) that puts items in AWS DynamoDB. We will also create a rule in [Amazon EventBridge](https://aws.amazon.com/eventbridge/) and another AWS Lambda that will retrieve data related to cost optimization from [AWS Compute Optimizer](https://aws.amazon.com/compute-optimizer) and [AWS Trusted Advisor](https://aws.amazon.com/trusted).
+ 
+When we complete our initial stage of template deployment, our deployed workload should reflect the following diagram:
+ 
+![Section2 Base Architecture](/watool/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/Images/section2/Architecture-Cost.png)
+ 
+Note the following:
+ 
+1. Amazon API Gateway has been provided and IAM role to invoke AWS Lambda function with mapping table that contains AWS Trusted Advisor Check IDs and Question ID of questions in Well-Architected Tool.
```
[the remark this answers is not in the document]
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md:18]: updated

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+ 
+2. AWS Lambda function has been provided and IAM role to put items in AWS DynamoDB.
+ 
+3. In Well-Architected Tool, a reviewer will define a workload which is a collection of resources and applications that delivers business value.
```
[the remark this answers is not in the document]
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md:22]: updated

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+ 
+Note the following:
+ 
+1. Amazon API Gateway has been provided and IAM role to invoke AWS Lambda function with mapping table that contains AWS Trusted Advisor Check IDs and Question ID of questions in Well-Architected Tool.
+ 
+2. AWS Lambda function has been provided and IAM role to put items in AWS DynamoDB.
+ 
+3. In Well-Architected Tool, a reviewer will define a workload which is a collection of resources and applications that delivers business value. 
+ 
+4. Defining a workload in Well-Architected Tool generates an event called **CreateWorkload** that Amazon EventBridge receives, which will invoke another AWS Lambda function.
```
[the remark this answers is not in the document]
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md:24]: updated

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+ 
+1. Amazon API Gateway has been provided and IAM role to invoke AWS Lambda function with mapping table that contains AWS Trusted Advisor Check IDs and Question ID of questions in Well-Architected Tool.
+ 
+2. AWS Lambda function has been provided and IAM role to put items in AWS DynamoDB.
+ 
+3. In Well-Architected Tool, a reviewer will define a workload which is a collection of resources and applications that delivers business value. 
+ 
+4. Defining a workload in Well-Architected Tool generates an event called **CreateWorkload** that Amazon EventBridge receives, which will invoke another AWS Lambda function. 
+ 
+5. This AWS Lambda function collects finding, reason, and recommended instance type for the rightsizing from **AWS Compute Optimizer**. It also gathers the details of "**Low Utilization Amazon EC2 Instances**" check from **AWS Trusted Advisor** such as Estimated Monthly Savings and Average CPU Utilization.
```
[the remark this answers is not in the document]
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md:26]: updated

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/_index.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+- **Phong Le**, Geo Solutions Architect, AWS Well-Architected
+ 
+## Contributor
+- **Ben Mergen**, Senior Cost Lead Solutions Architect, AWS Well-Architected
+- **Stephen Salim**, Senior Geo Solutions Architect, AWS Well-Architected
+ 
+## Introduction
+ 
+AWS Well-Architected Framework Review is a constructive conversation about architectural decisions. One of the best ways to maximize the impact of the review is to be equipped with the preliminary review analysis data points. Once data is available for the review, the reviewer can specifically ask pillar-aligned questions, and customers can authoritatively answer those questions based on the data. In this lab, we will focus on **cost optimization** review utilizing insights to support **Rightsizing** decision-making to align the size of your service to your actual workload demand. This allows customers to see an immediate cost reduction benefit using insights from the data. The purpose of this lab is to walk you through one of many integration examples with Well-Architected Tool to conduct a data-driven Cost Optimization review using [AWS Cloud Financial Management Services](https://aws.amazon.com/aws-cost-management/).
```
[the remark this answers is not in the document]
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/_index.md:20]: updated

[the remark this answers is not in the document]
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/_index.md:20]: updated

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+```
+* Answer **y** for LambdaPutDynamoDB may not have authorization defined, Is this ok?
+ 
+![Section2 SAMDeploy](/watool/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/Images/section2/SAMDeploy.png)
+ 
+3. In Outputs, take a note of **APIGWUrl**.
+ 
+![Section2 APIGWUrl](/watool/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/Images/section2/APIGWUrl.png)
+ 
+4. Now we will update the AWS DynamoDB table with a sample mapping table in json file through API Gateway. This mapping table has Question ID of Well-Architected question associated with AWS Trusted Advisor check ID and AWS Lambda function would retrieve Question ID to update findings related to cost optimization into notes in Well-Architected Tool.
```
[the remark this answers is not in the document]
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md:97]: updated

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+ 
+5. Confirm that UnprocessedItems appear to be empty, which means you successfully put items into AWS DynamoDB. 
+![Section2 Confirm](/watool/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/Images/section2/Confirm.png)
+ 
+6. In AWS DynamoDB console, click **wa-mapping** you just deployed and click **Explore table items**. 
+![Section2 Table](/watool/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/Images/section2/Table.png)
+ 
+![Section2 Explore](/watool/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/Images/section2/Explore.png)
+ 
+7. There are 1 Question IDs of Well-Architected questions and 1 AWS Trusted Advisor checks.
```
[the remark this answers is not in the document]
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md:115]: updated

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/3_create_workload.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
+---
+title: "Create a Well-Architected Workload with Tags"
+date: 2020-12-17T11:16:09-04:00
+chapter: false
+weight: 3
+pre: "<b>3. </b>"
+---
+ 
+## Overview
+Well-Architected Framework Reviews are conducted per workload. A workload is a collection of resources and code that delivers business value, such as a customer-facing application or a backend process. Let’s define a workload.
```
[the remark this answers is not in the document]
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/3_create_workload.md:10]: updated

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/3_create_workload.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+ 
+3. When AWS lambda function retrieves cost optimization data, it will validate if Amazon EC2 Instances are being used for the particular workload that you defined in Well-Architected tool using Amazon Tags before ingesting data points into notes in Well-Architected Tool. Therefore, a reviewer can focus on EC2 Instances used only for the particular workload. Attach the same tags we have attached to Amazon EC2 Instances to this workload. Use **workload** as Key and **wademo** as Value.
+AWS Lambda functions will verify if cost optimization data from AWS Compute Optimizer and AWS Trusted Advisor has the same tags as the following tags to ensure Amazon EC2 Instances are being used for this workload.
+    ```
+    Key = workload
+    Value = wademo
+    ```
+![Section3 Tags](/watool/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/Images/section3/Tags.png)
+ 
+4. Choose the lenses that apply to this workload. **AWS Well-Architected Framework** has been selected by default.
```
[the remark this answers is not in the document]
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/3_create_workload.md:36]: updated

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/4_perform_review.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
+---
+title: "Performing a data-driven review"
+date: 2020-12-17T11:16:09-04:00
+chapter: false
+pre: "<b>4. </b>"
+weight: 4
+---
+ 
+![Section4 Integration](/watool/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/Images/section4/TA_Aco_integration.png)
```
[the remark this answers is not in the document]
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/4_perform_review.md:9]: updated

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/1_prerequisites.md (hunk 1 above)
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/1_prerequisites.md]: took a new screenshot and replaced

**Comment by awswa** [comment]
> 

I made changes that reviewers addressed

**Review by awswa — APPROVED** [review]
I made changes that reviewers addressed

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/1_prerequisites.md (hunk 1 above)
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/1_prerequisites.md]: > Will it make more sense to prescribe the tag ?
> 
> ```
>    Key = workload
>    Value = wademo
>    ```
> ```

took a new screenshot and replaced

File: content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (awswa) [reply content/Well-ArchitectedTool/200_Labs/200_Integration_with_AWS_Compute_Optimizer_and_AWS_Trusted_Advisor/2_configure_env.md]: > Our SAM deployment includes an Amazon API gateway with an IAM role to invoke our Lambda function. This function references a JSON mapping file that contains trusted advisor check IDs allowing for mapping to the appropriate questions within the Well-Architected tool.

update!

**Review by awswa — COMMENTED** [review]
I have made changes based on comments from reviewers.

**Review by awswa — APPROVED** [review]
I have completed all changes and it's ready to be published.

**Review by sssalim-aws — APPROVED** [review]
Review Completed by sssalim & tearatim, approving change.

**Pull request merged**